api/webapp.go

```diff
@@ -20,20 +21,25 @@ func (s *Server) ServeWebapp() {
 	}
 
 	m := http.NewServeMux()
-	m.Handle("/", s.middleware(s.WebappHandler))
-	m.Handle("/webapp.js", s.WebappJSHandler())
+	m.Handle("/", s.middleware(s.WebappTemplateHandler))
+	m.Handle("/webapp/", s.FrontendHandler("/webapp"))
+
 	webappserver := &http.Server{Handler: m}
 	webappserver.Serve(listener)
 	return
 }
 
-// WebappJSHandler attempts to resolve an update
-func (s *Server) WebappJSHandler() http.Handler {
+// FrontendHandler resolves the current webapp hash, (fetching the compiled frontend in the process)
+// and serves it up as a traditional HTTP endpoint, transparently redirecting requests
+// for [prefix]/foo.js to [CAFS Hash]/foo.js
+// prefix is the path prefix that should be stripped from the request URL.Path
+func (s *Server) FrontendHandler(prefix string) http.Handler {
 	var webappPath = s.cfg.Webapp.EntrypointHash
 	go s.resolveWebappPath(&webappPath)
```

Comment:
I'm not a fan of this line for multiple reasons: it's confusing, it's mutating its input parameter, it has a race condition with the code below, there's no error handling, and really the data could be owned by `Server` and initialized by it, as there's nothing that necessitates it being within FrontendHandler. Could you add a comment explaining some of what's going on, or maybe add a TODO to fix it later?

Reply:
Totally agreed. Don't have time to fix now, but I've documented your comments here in a TODO